dynamiqs/time_array.py

```diff
@@ -673,6 +723,14 @@ def dtype(self) -> jnp.dtype:
     def shape(self) -> tuple[int, ...]:
         return jnp.broadcast_shapes(*[tarray.shape for tarray in self.timearrays])
 
+    @property
+    def dims(self) -> tuple[int, ...]:
+        return self.timearrays[0].dims
+
+    @property
+    def ndiags(self) -> int:
```

Comment:
`SummedTimeArray` accepts mixed number of diags so it's ill defined

Reply:
Not sure it's ill-defined, since when summing sparse qarrays, the final result will also be a sparse qarray with a well-defined `ndiags`. This is what we should return.

However, it is ill-defined if one sub-array is dense, but it's the same for other `TimeQArray` types.